controllers/gatling_controller.go

```diff
@@ -373,7 +373,7 @@ func doNotRequeue(err error) (ctrl.Result, error) {
 
 func (r *GatlingReconciler) createVolumesForCR(ctx context.Context, gatling *gatlingv1alpha1.Gatling, namespace string, log logr.Logger) error {
 	// Create Simulation Data ConfigMap if defined to create in CR
-	if &gatling.Spec.TestScenarioSpec.SimulationData != nil && len(gatling.Spec.TestScenarioSpec.SimulationData) > 0 {
+	if gatling.Spec.TestScenarioSpec.SimulationData != nil && len(gatling.Spec.TestScenarioSpec.SimulationData) > 0 {
```

Comment:
When gatling is nil, the nil pointer access is panicked, so here is the case
```
if gatling ! = nil && gatling.Spec.TestScenarioSpec.SimulationData && len(gatling.Spec.TestScenarioSpec.SimulationData) > 0
```
is a good choice here.
The zero value of map is nil, so the nil check of SimulationData is left as it is.

Reply:
Thank you for the review!

It is indeed a very good idea. 
However, I thought that using this approach might change the existing behavior if gatling is nil.
For example, if gatling is nil in the createVolumesForCR function, no processing is performed and error is returned as nil.

To prevent this, I think it would be better to check for nil at the beginning of the function and return an error at that point.
What do you think?

Here's what I have in mind to add at the beginning of the function.
```go
if gatling == nil {
    return errors.New("gatling is nil")
}
```

Reply:
@kane8n 
Sorry for the delay in making the corrections.
I have addressed it with https://github.com/st-tech/gatling-operator/pull/126/commits/60bbeb037e3edd072ab0e660d05b2b43a2a239e9.
Please check it.